Source/Csla/Core/GraphMerger.cs

```diff
@@ -202,5 +202,173 @@ private void MergeGraph(IEditableCollection target, IEditableCollection source)
+            }
+          }
+          else
+          {
+            if (sourceValue is IEditableCollection sourceList)
+            {
+              var targetFieldExists = true;
+              if (targetFieldManager != null && (item.RelationshipType & RelationshipTypes.LazyLoad) == RelationshipTypes.LazyLoad)
+                targetFieldExists = targetFieldManager.FieldExists(item);
+              if (targetFieldExists && ReadProperty(target, item) is IEditableCollection targetList)
+              {
+                await MergeGraphAsync(targetList, sourceList);
+              }
+              else
+              {
+                if ((item.RelationshipType & RelationshipTypes.PrivateField) == RelationshipTypes.PrivateField)
+                  Reflection.MethodCaller.CallPropertySetter(target, item.Name, sourceList);
+                else
+                  await Task.Run(() => LoadProperty(target, item, sourceList));
+              }
+            }
+            else
+            {
+              if ((item.RelationshipType & RelationshipTypes.PrivateField) == RelationshipTypes.PrivateField)
+                Reflection.MethodCaller.CallPropertySetter(target, item.Name, sourceValue);
+              else if (sourceValue != null || (item.RelationshipType & RelationshipTypes.LazyLoad) != RelationshipTypes.LazyLoad)
+                await Task.Run(() => LoadProperty(target, item, sourceValue));
+            }
+          }
+        }
+        if (source.IsNew)
+        {
+          await Task.Run(() => MarkNew(target));
+        }
+        else if (!source.IsDirty)
+        {
+          await Task.Run(() => MarkOld(target));
+        }
+        else
+        {
+          CopyField(source, target, "_isDirty");
+          CopyField(source, target, "_isNew");
+          CopyField(source, target, "_isDeleted");
+        }
+        await CheckRulesAsync(target);
+      }
+    }
+    /// <summary>
+    /// Merges state from source graph into target graph.
+    /// </summary>
+    /// <param name="target">Target of merge.</param>
+    /// <param name="source">Source for merge.</param>
+    private async Task MergeGraphAsync(IEditableCollection target, IEditableCollection source)
+    {
+      var listType = await Task.Run(() => target.GetType());
+      var childType = await Task.Run(() => Utilities.GetChildItemType(listType));
+      var genericTypeParams = new Type[] { listType, childType };
+      System.Reflection.MethodInfo methodReference;
+      if (typeof(IExtendedBindingList).IsAssignableFrom(listType))
+        methodReference = await Task.Run(() => GetType().GetMethod("MergeBusinessBindingListGraphAsync"));
+      else
+        methodReference = await Task.Run(() => GetType().GetMethod("MergeBusinessListGraphAsync"));
+      var gr = await Task.Run(() => methodReference.MakeGenericMethod(genericTypeParams));
+      await Task.Run(() => gr.Invoke(this, [target, source]));
```

Comment:
This invocation must be proper async. The current implementation does nto wait for `gr.Invoke()` to complete.

Reply:
noted

Reply:
Fixed. Please check.